### ISSUE TITLE
Why not call -sd_removeImageLoadOperationWithKey: after UIView+WebCache call completion block

### ISSUE THREAD

**Issue by Thememont** [rs/SDWebImage#2125]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | ios
 Platform Version        | 11.2
 SDWebImage Version      | 4.2.2
 Integration Method      | cocoapods
 Xcode Version           | Xcode 9.2
 Repro rate              | all the time (100%) 

### Issue Description and Steps

I found that the UIView webCache extensions will retain the objects in completion block. So why not call -sd_removeImageLoadOperationWithKey: after UIView+WebCache call completion block ?

```
// UIView+WebCache.m   line 74 ~ 84
SDWebImageNoParamsBlock callCompletedBlockClojure = ^{
    if (!sself) { return; }
    if (!shouldNotSetImage) {
        [sself sd_setNeedsLayout];
    }
    if (completedBlock && shouldCallCompletedBlock) {
        completedBlock(image, error, cacheType, url);
    }
                
    [self sd_removeImageLoadOperationWithKey:validOperationKey];  // add this line
};
```
It can break the retain cycle so that we needn't use weak - strong dance at completion block. 

Unit tests all pass after add this code.

**Comment by dreampiggy** [comment]
It seems a design issue(bug). This code logic is not changed from SD 4.0. I guess retain the previous operation can help to check if current operation has done and do some logic in previous version SD(perhaps 3.x). But current implementation do not expose any public interface to get that operation, just keep in a NSMutableDictionary and keep alive with the UIView instance. This cost unnecessary memory

Maybe we should fix this, could you please provide a PR ?

**Comment by Thememont** [comment]
@dreampiggy provided

**Comment by dreampiggy** [comment]
Fixed via #2132
